contrib/packs/actions/pack_mgmt/get_pack_dependencies.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2019 Extreme Networks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import six
+
+from st2common.constants.pack import PACK_VERSION_SEPARATOR
+from st2common.content.utils import get_pack_base_path
+from st2common.runners.base_action import Action
+from st2common.util.pack import get_pack_metadata
+
+STACKSTORM_EXCHANGE_PACK = 'StackStorm-Exchange/stackstorm-'
+LOCAL_FILE_PREFIX = 'file://'
+
+
+class GetPackDependencies(Action):
+    def __init__(self, config=None, action_service=None):
+        self.dependency_list = []
+        self.conflict_list = []
+
+    def run(self, packs_status, nested):
+        """
+        :param packs_status: Name of the pack in Exchange or a git repo URL and download status.
+        :type: packs_status: ``dict``
+
+        :param nested: Nested level of dependencies to prevent infinite or really
+        long download loops.
+        :type nested: ``integer``
+        """
+        result = {}
+
+        if not packs_status or nested == 0:
+            return result
+
+        for pack, status in six.iteritems(packs_status):
+            if 'success' not in status.lower():
+                continue
+
+            dependency_packs = get_dependency_list(pack)
+            if not dependency_packs:
+                continue
+
+            for dependency_pack in dependency_packs:
+                pack_and_version = dependency_pack.split(PACK_VERSION_SEPARATOR)
+                name_or_url = pack_and_version[0]
+                pack_version = pack_and_version[1] if len(pack_and_version) > 1 else None
+
+                if name_or_url.startswith(LOCAL_FILE_PREFIX):
```

Comment:
It's optimistic thinking that destination pack format can only consist of 3 examples like filesystem, pack exchange and git http URL.

There are much more: 
https://github.com/StackStorm/st2/blob/26ffc2f4fe16b7a5b6fe46b98e32daf88e473dde/contrib/packs/tests/test_action_download.py#L460-L491

As said before pack dependency format should support the same formats that `st2 pack install` does.
This means instead of code duplication there should be one source of truth and common functionality every pack install component can refer to.

Reply:
Thanks for point out example. Will rework on parse pack input.